### ISSUE TITLE
User actions sequence classification in Flair

### ISSUE THREAD

**Issue by klimentij** [zalandoresearch/flair#1317]
I feel it's possible to apply your awesome library for this tricky non-NLP task. I've read all your tutorials, but I'm still not sure if it will work out and how it's better to implement it.  

I have a training set where each row is a series of user actions on a website (logged in, sent an invoice, etc.) and times deltas in ms between these actions. Each row has a label — a corresponding user class (10 classes):

```
action132, 2340, action21, 300, ...  3
action238, 1240, action22, 350, ...  6
...
action763, 1240, action42, 750, ...  2 
```

Say, I have 500 unique action types and each row has a variable length. Max sequence length is 100k and there are 10k rows in the training set (10k users).

I need to predict labels for the test set.

How could I use Flair TextClassifier to solve this problem? I guess I'll have to convert milliseconds to bins so that all features are categorical (like words in some language) and then I'll have to somehow create a model for this new "language" and then apply TextClassifier as your tutorial shows, right?

**Comment by pommedeterresautee** [comment]
just say that each action is a token... and that's all. Not sure Flair is the best choice for classif with such dataset as you won't use any pretrained LM (don't know if order count in those actions, if not fasttext would be a better starting point).

**Comment by klimentij** [comment]
Yes, the order of actions is important. Should I train some LM from scratch then? If so, which one to use?

**Comment by pommedeterresautee** [comment]
You need a lm that is word based. Bert and sons are subword based. Flair is character based. An idea is to train your own embeddings (fasttext for instance) and use only this representation. The sequence will be managed by the crf layer.

**Comment by klimentij** [comment]
Great, thank you!
